### ISSUE TITLE
assigning a TypeVar to an identical value is reported as an error.

### ISSUE THREAD

**Issue by mrolle45** [python/mypy#12528]
In my project, I have multiple definitions of the same TypeVar.  They originated from merging (with pytype) the original modules with their generated stub files.  It may be that pytype was in error, but I'll look into that later.
In this example, the TypeVar `Self` appears several times in the stub files.  The result is:
```
A.py:
from B import *
Self = TypeVar('Self')

B.py:
Self = TypeVar('Self')
```
mypy reports the redefinition of `Self` in `A` as an error.

I expect that a type checker should consider this as a benign redefinition.  It would ignore it, and replace the existing symbol table node (called `B.Self`) with a new node called `A.Self`.  On any subsequent reanalysis of `A`, I believe that mypy would treat this as already analyzed and do nothing.

You might disagree with me about this.  Does the PEP have any specification about this?

In `semanal.py` line 3157 is the fail message.  I would say that the existing node should first be compared with the new name, values, and bound for equality.  When they are equal, skip the subsequent fail message and the return of False.

**Your Environment**

- Mypy version used: 0.931
- Python version used: 3.7

**Comment by mrolle45** [comment]
Here's a patch I made to `semanal.py`:
```
3135,3137c3135,3136
<             if [upper_bound] + values != [existing.node.upper_bound] + existing.node.values:
<                 self.fail('Cannot redefine "%s" as a type variable "%s"' % (existing.fullname, name), s)
<                 return False
---
>             self.fail('Cannot redefine "%s" as a type variable' % name, s)
>             return False
```
Note that I also improved the error message, to show where the original TypeVar came from, when it's a real change.

**Comment by mrolle45** [comment]
Correction to the patch.  My test crashed if `existing.node` was not a `TypeVarExpr`.
```
3135,3139c3135,3136
<             if (not isinstance(existing.node, TypeVarExpr)
<                 or [upper_bound] + values != [existing.node.upper_bound] + existing.node.values
<                 ):
<                 self.fail('Cannot redefine "%s" as a type variable "%s"' % (existing.fullname, name), s)
<                 return False
---
>             self.fail('Cannot redefine "%s" as a type variable' % name, s)
>             return False
```
